File: ccc08s1.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    char city[257];
    int minTemp = INT_MAX;
    char minCity[257];
    do {
        int temp;
        scanf("%s %d", &city[0], &temp);
        if (temp < minTemp) {
            minTemp = temp;
            strcpy(minCity, city);
        }
    } while (strcmp(city, "Waterloo") != 0);
    printf("%s", minCity);
    return 0;
}

File: ccc08s1.java
import java.util.Scanner;

public class ccc08s1 {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        String maxCity = "";
        int minTemp = Integer.MAX_VALUE;
        while (true) {
            String[] line = in.nextLine().split(" ");
            String city = line[0];
            int temp = Integer.parseInt(line[1]);
            if (temp < minTemp) {
                maxCity = city;
                minTemp = temp;
            }
            if (city.equals("Waterloo")) {
                break;
            }
        }
        System.out.println(maxCity);
    }

}
